nemo/collections/nlp/modules/common/megatron/transformer.py

```diff
@@ -1319,7 +1320,9 @@ def forward(
 
             layernorm_input = bias_dropout_add_func(attention_output, attention_bias, residual, self.hidden_dropout)
             normalization_output = self.post_inter_attention_layernorm(layernorm_input)
-
+            # Post-LN normalization after residual
+            if self.transformer_block_type == 'post_ln':
+                layernorm_input = normalization_output
```

Comment:
Why not set directly `residual` here? why set `layernorm_input`?

Reply:
because later logic set `residual = layernorm_input`